File: programmers/src/main/kotlin/네트워크.kt
fun main(){
    class Solution {
        fun solution(n: Int, computers: Array<IntArray>): Int {
            var answer = 0
            val visited = BooleanArray(n){false}
            fun dfs(visiting:Int){
                visited[visiting] = true
                for(i in 0 until n){
                    if(i!=visiting && !visited[i] && computers[visiting][i]==1){
                        dfs(i)
                    }
                }
            }
            for(i in 0 until n){
                if(!visited[i]){
                    answer++
                    dfs(i)
                }
            }
            return answer
        }
    }
}

File: boj/네트워크.py
def solution(n, computers):
    answer = 0
    visited = [False for i in range(0, n)]

    def dfs(visiting):
        visited[visiting] = True
        for i in range(0, n):
            if i != visiting and not visited[i] and computers[visiting][i] == 1:
                dfs(i)

    for i in range(0, n):
        if not visited[i]:
            answer += 1
            dfs(i)

    return answer


print(solution(3, [[1, 1, 0], [1, 1, 1], [0, 1, 1]]))
